src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -25,17 +25,404 @@ namespace XBU = XBUtilities;
+  // May need to refactor this when adding functionality for multiple devices
+  std::cout << boost::format("%s : %d\n") % "Device BDF" % flasher.sGetDBDF();
+  std::cout << "Current Configuration\n";
+
+  std::cout << boost::format("  %-20s : %s\n") % "Platform" % currentDSA.name;
+  std::cout << boost::format("  %-20s : %s\n") % "SC Version" % currentDSA.bmcVer;
+  std::cout << boost::format("  %-20s : 0x%x\n") % "Platform ID" % currentDSA.timestamp;
+
+  std::cout << "\nIncoming Configuration\n";
+  std::pair <std::string, std::string> s = deployment_path_and_filename(candidate.file);
+  std::cout << boost::format("  %-20s : %s\n") % "Deployment File" % s.first;
+  std::cout << boost::format("  %-20s : %s\n") % "Deployment Directory" % s.second;
+  std::cout << boost::format("  %-20s : %s\n") % "Size" % file_size(candidate.file.c_str());
+  std::cout << boost::format("  %-20s : %s\n\n") % "Timestamp" % get_file_timestamp(candidate.file.c_str());
+
+  std::cout << boost::format("  %-20s : %s\n") % "Platform" % candidate.name;
+  std::cout << boost::format("  %-20s : %s\n") % "SC Version" % candidate.bmcVer;
+  std::cout << boost::format("  %-20s : 0x%x\n\n") % "Platform ID" % candidate.timestamp;
+
+  std::cout << "----------------------------------------------------\n";
+  std::cout << "Actions to perform:\n";
+  if(!same_dsa)
+    std::cout << "  -Program flash image\n";
+  if(!same_bmc)
+    std::cout << "  -Program SC image\n";
+  std::cout << "----------------------------------------------------\n";
+  // end header info
+
+  return candidate;
+}
+
+/* 
+ * Confirm with the user
+ * Helper method for auto_flash
+ */
+bool canProceed()
+{
+  std::string input;
+  bool answered = false;
+  bool proceed = false;
+
+  while (!answered) {
+    std::cout << "Are you sure you wish to proceed? [y/n]: ";
+    std::cin >> input;
+    if(input.compare("y") == 0 || input.compare("n") == 0)
+      answered = true;
+  }
+
+  proceed = (input.compare("y") == 0);
+  if (!proceed)
+    std::cout << "Action canceled." << std::endl;
+  return proceed;
+}
+
+/* 
+ * Flash shell and sc firmware
+ * Helper method for auto_flash
+ */
+static int 
+updateShellAndSC(uint16_t boardIdx, DSAInfo& candidate, bool& reboot)
+{
+  reboot = false;
+
+  Flasher flasher(boardIdx);
+  if(!flasher.isValid())
+    throw xrt_core::error(boost::str(boost::format("%d is an invalid index") % boardIdx));
+
+  bool same_dsa = false;
+  bool same_bmc = false;
+  DSAInfo current = flasher.getOnBoardDSA();
+  if (!current.name.empty()) {
+    same_dsa = (candidate.name == current.name &&
+      candidate.matchId(current));
+    same_bmc = (current.bmcVer.empty() ||
+      candidate.bmcVer == current.bmcVer);
+  }
+  if (same_dsa && same_bmc)
+    std::cout << "update not needed" << std::endl;
+
+  if (!same_bmc) {
+    std::cout << "Updating SC firmware on card[" << flasher.sGetDBDF() <<
+      "]" << std::endl;
+    auto ret = 0;
+    update_SC(boardIdx, candidate.file.c_str());
```

Comment:
This is a general comment for the entire PR.  Please do not use .c_str() to pass to a function that takes const std::string& (or std::string for that matter).  Please revisit all .c_str() and check if they truly are needed.  I checked this one, and it is not needed.